classes/migration/upgrade/v3_5_0/I9253_SiteAnnouncements.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_5_0/I9253_SiteAnnouncements.php
+ *
+ * Copyright (c) 2014-2023 Simon Fraser University
+ * Copyright (c) 2000-2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I9253_SiteAnnouncements
+ */
+
+namespace PKP\migration\upgrade\v3_5_0;
+
+use APP\core\Application;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use PKP\migration\Migration;
+
+class I9253_SiteAnnouncements extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('announcement_types', function (Blueprint $table) {
+            $table->dropForeign(['context_id']);
```

Comment:
This would definitely need to be aligned with https://github.com/pkp/pkp-lib/pull/9076#discussion_r1228787990; just dropping the foreign key constraint is a step backwards. In the database layer, I'm 100% sure we'll want to use `null` in place of the previous magic number `0` for site-wide; it's just the PHP layer that still needs the wrinkles worked out.

Reply:
Ok, that might remove the need for a migration as well.

Reply:
That's fine, it's necessarily messy until we have a consistent pattern. IIRC there's somewhere else where we allow `[CONTEXT_ID_NONE]` as a list of context IDs, but all this will need to be grepped and refactored as part of https://github.com/pkp/pkp-lib/pull/9076. Please carry on!